p5js/FaceApi/FaceApi_Image_Landmarks/sketch.js

```diff
@@ -0,0 +1,116 @@
+let faceapi;
+let img;
+let detections;
+
+// by default all options are set to true
+const detection_options = {
+    withFaceLandmarks: true,
+    withFaceExpressions: false,
+    withFaceDescriptors: false,
+}
+
+function preload(){
+    img = loadImage('assets/frida.jpg')
+}
+
+function setup() {
+    createCanvas(200, 200);
+    img.resize(width, height);
+
+    faceapi = ml5.faceApi(detection_options, modelReady)
+    textAlign(RIGHT);
+}
+
+function modelReady() {
+    console.log('ready!')
+    console.log(faceapi)
+    faceapi.detectSingle(img, gotResults)
+
+}
+
+function gotResults(err, result) {
+    if (err) {
+        console.log(err)
+        return
+    }
+    // console.log(result)
+    detections = result;
+
+    // background(220);
+    background(255);
+    image(img, 0,0, width, height)
+    if (detections) {
+        // console.log(detections)
+        drawBox(detections)
+        drawLandmarks(detections)
+    }
+}
+
+function drawBox(detections){
+    const {alignedRect} = detections;
+    const {_x, _y, _width, _height} = alignedRect._box;
+    noFill();
+    stroke(255, 0, 0)
+    strokeWeight(2)
+    rect(_x, _y, _width, _height)
+}
+
+function drawLandmarks(detections){
+    const {landmarks} = detections;
+    const mouth = landmarks.getMouth();
+    const nose = landmarks.getNose();
+    const leftEye = landmarks.getLeftEye();
+    const leftEyeBrow = landmarks.getLeftEyeBrow();
+    const rightEye = landmarks.getRightEye();
+    const rightEyeBrow = landmarks.getRightEyeBrow();
+    
+    noFill();
+    stroke(0, 0, 0)
+    strokeWeight(2)
+    
+    push()
+    // mouth
+    beginShape();
+    mouth.forEach(item => {
+        vertex(item._x, item._y)
```

Comment:
Same comment re: underscore. If we are encouraging users of ml5 to access these properties I would lose the underscore probably?

Reply:
@shiffman - the underscore _x and _y are part of the way face-api.js returns the data. We could go through and update the "_x" and "_y" in the returns? I worry about adding additional computation as well as changing the data structure too much in case people are using face-api in other contexts. Should we keep the  "_x" and "_y" or rather change it?

Reply:
Ah, I see, I didn't realize this was the face API directly! I think we can keep it as is and maybe file this as something to think about later if it's causing confusion or we end up wrapping elements of the API for other reasons?

Reply:
@shiffman - I wonder now if maybe it just makes sense to include in the return:

```
landmarkFeatures:
{
mouth: [array of {_x, _y}]
nose:[array of {_x, _y}]
leftEye:[array of {_x, _y}]
rightEye: [array of {_x, _y}]
...etc
}
```

Reply:
Oh, I like this idea! Very similar to PoseNet actually!

Reply:
Cool! I'll add this in now :)

Reply:
What if we just call it `landmarks`? or `features`?

Reply:
Since `landmarks` is the name of the returned results of the landmarks, I'd go for maybe just "parts" ?